guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/GuavaBeanSerializerModifier.java

```diff
@@ -24,7 +28,12 @@ public List<BeanPropertyWriter> changeProperties(SerializationConfig config,
     {
         for (int i = 0; i < beanProperties.size(); ++i) {
             final BeanPropertyWriter writer = beanProperties.get(i);
-            if (Optional.class.isAssignableFrom(writer.getType().getRawClass())) {
+            if (writer instanceof UnwrappingBeanPropertyWriter) {
```

Comment:
```suggestion
            // [databind#0000] Since 2.17 : problem to solve + reason for matching Unwrapping....Writer
            if (writer instanceof UnwrappingBeanPropertyWriter) {
```

Commenting why this fix was made by unwrapping writer would help.